pallets/attestation/src/lib.rs

```diff
@@ -348,9 +361,97 @@ pub mod pallet {
 				},
 			);
 
-			Self::deposit_event(Event::AttestationRevoked(revoker, claim_hash));
+			Self::deposit_event(Event::AttestationRevoked(attester, claim_hash));
+
+			Ok(Some(<T as pallet::Config>::WeightInfo::revoke(delegation_depth)).into())
+		}
+
+		/// Remove an attestation.
+		///
+		/// The origin must be either the creator of the attestation being or
+		/// an entity that in the delegation tree is an ancestor of the
+		/// attester, i.e., it was either the delegator of the attester or an
+		/// ancestor thereof.
+		///
+		/// Emits `AttestationRemoved`.
+		///
+		/// # <weight>
+		/// Weight: O(P) where P is the number of steps required to verify that
+		/// the dispatch Origin controls the delegation entitled to revoke the
+		/// attestation. It is bounded by `max_parent_checks`.
+		/// - Reads: [Origin Account], Attestations, delegation::Roots
+		/// - Reads per delegation step P: delegation::Delegations
+		/// - Writes: Attestations, DelegatedAttestations
+		/// # </weight>
+		#[pallet::weight(<T as pallet::Config>::WeightInfo::remove(*max_parent_checks))]
+		pub fn remove(
+			origin: OriginFor<T>,
+			claim_hash: ClaimHashOf<T>,
+			max_parent_checks: u32,
+		) -> DispatchResultWithPostInfo {
+			let source = <T as Config>::EnsureOrigin::ensure_origin(origin)?;
+			let attester = source.subject();
```

Comment:
S: Same as above.

Reply:
The origin is often called who so I renamed it to who 50c9545